src/FileServiceManager/FileService/DAL/FileSystemConfiguration/FileSystemJsonConfiguration/FileSystemJsonConfigurator.cs

```diff
@@ -0,0 +1,229 @@
+﻿using FileService.Models;
+using Microsoft.Extensions.Configuration;
+using Models;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NLog.Fluent;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+
+/// <summary>
+/// FileSystemJsonConfigurator implementing IFileSystemConfigurator. Used to read/write/remove json configuration about company's file system structure
+/// </summary>
+
+namespace FileService.DAL
+{
+    public class FileSystemJsonConfigurator : IFileSystemConfigurator
+    {
+        private readonly string _configPath = "";
+
+        public FileSystemJsonConfigurator(IConfiguration configuration)
+        {
+            _configPath = Path.Combine(configuration["storage:path"] ?? Directory.GetCurrentDirectory(), "FileSystemConfig.json");
+        }
+
+        public async Task AddConfigurationAsync(int companyId, FileMetadata fileMetadata)
+        {
+            JToken rootToken = null;
+            StreamReader fileReader = null;
+            StreamWriter fileWriter = null;
+            try
+            {
+                using (fileReader = File.OpenText(_configPath))
+                {
+                    using (JsonTextReader reader = new JsonTextReader(fileReader))
+                    {
+                        rootToken = await JToken.ReadFromAsync(reader);
+                    }
+                }
+
+                if (rootToken != null)
```

Comment:
This class needs to be covered by Unit Tests

Reply:
This is on Hrach's side. He is working on it